### PR TITLE
Spire proposal/testing results doc

### LINKED ISSUE #436: Discover feasibility for using SPIRE/SPIFFE in tarmak
Discover feasibility for using SPIRE/SPIFFE in tarmak to strengthen node indentity, investigation and if appropiate write a proposa

/assign @JoshVanL

### PR DESCRIPTION
**Pull request by JoshVanL**
**What this PR does / why we need it**:
This adds a document detailing the feasibility of adding spire into the Tarmak cluster. It includes steps needed to achieve this and results of some testing.

fixes #436

```release-note
SPIFFE/SPIRE proposal/feasibility document.
```

/assign @simonswine

### CODE REVIEW

File: docs/proposals/spiffe-spire-integration.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+primitives, cloud provider meta data and even Kubernetes primitives. Used in
+conjunction, multiple identifiers provide a greater amount of security within
+verifiable, authentication of identity, not just a single leak-able token.
+
+Objective
+---------
+
+Use Spire within the Tarmak cluster to bolster the authentication of workload
+identities, resulting in a more secure cluster. This could ultimately mean a
+large or complete replacement of Vault and it's periphery tools, in favour of
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply docs/proposals/spiffe-spire-integration.rst]: Yes exactly, think that could be a little better written^ !
I'm am concerned however about how much code we need to touch - if any - in the security path, given spire's maturity. Need to weigh whether it's worth adding the extra layer of complexity too.

File: docs/proposals/spiffe-spire-integration.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+Spire is not currently well built for HA and really only expects a single server
+in cluster. PostgreSQL support for HA is quite poor from what I have found,
+whilst also being the only supported "HA" backend. I was able to use a tool Patroni_ that
+facilitates a PostgreSQL cluster with Consul however only the master is in write
+mode. All replicas on other instances are in read mode which causes spire severs
+on those instances to panic. Spire servers have to connect to the PostgreSQL
+cluster master elected through Consul.
+
+Spire uses Elliptic Curve keys which is odd and can be annoying.
```
Comment (simonswine) [inline docs/proposals/spiffe-spire-integration.rst]: Not too sure either provide more evidence or get rid of this sentence. There are good reasons IMHO for ECC (https://blog.cloudflare.com/ecdsa-the-digital-signature-algorithm-of-a-better-internet/)

**Comment by simonswine** [comment]
/assign @JoshVanL 
/unassign 

I think james' comment is quite valid. Also the one about the ECC and last but not least DCO

**Pull request merged**